File: server/src/main/java/com/wecp/healthcare_appointment_management_system/entity/Doctor.java
package com.wecp.healthcare_appointment_management_system.entity;

import javax.persistence.*;
import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import java.util.HashSet;
import java.util.Set;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "doctors")
public class Doctor extends User
{
   private String specialty;

   private String availability;

   @OneToMany(mappedBy = "doctor", cascade = CascadeType.ALL)
   @JsonIgnore
   private Set<Appointment> appointments = new HashSet<>();

   @OneToMany(mappedBy = "doctor", cascade = CascadeType.ALL)
   @JsonIgnore
   private Set<MedicalRecord> medicalRecords = new HashSet<>();
   
}

File: client/src/types/Doctor.ts
import { Appointment } from "./Appointment";
import { MedicalRecord } from "./MedicalRecord";

export interface Doctor
{
    id?: number;
    username: string;
    password: string;
    email: string;
    role: string;
    specialty: string;
    availability : string;
    appointments: Appointment;
    medicalRecords: MedicalRecord;

}